### PR TITLE
Massive cleanup add ranking

### PR DESCRIPTION
**Pull request by reifcode**
Add ranking and score availability for Activities and related tests

### CODE REVIEW

File: activity_test.go [hunk 1]
```diff
@@ -116,3 +118,40 @@ func TestActivityUnmarshallBadPayloadTo(t *testing.T) {
...
 }
+
+func TestActivityUnmarshalScore(t *testing.T) {
+	activity := &getstream.Activity{}
+
+	expected := 1.123
+	payload := []byte(fmt.Sprintf(`{ "score": %f }`, expected))
+
+	err := activity.UnmarshalJSON(payload)
+	if err != nil {
```
Comment (JelteF) [inline activity_test.go]: Can you use testify in the new tests you write? Both here and below.

File: feed_flat_test.go [hunk 2]
```diff
@@ -2,14 +2,19 @@ package getstream_test
...
 	getstream "github.com/GetStream/stream-go"
 )
 
-func getFlatFeed(t *testing.T, client *getstream.Client) *getstream.FlatFeed {
-	feed, err := client.FlatFeed("flat", RandString(8))
+const (
+	flatFeedName = "flat"
+)
+
+func getFlatFeed(t *testing.T, client *getstream.Client, name string) *getstream.FlatFeed {
```
Comment (JelteF) [inline feed_flat_test.go]: Can you just make 2 functions, getFlatFeed and getRankedFeed. Then all the calls can just stay the same.

File: activity_test.go (hunk 1 above)
Comment (reifcode) [inline activity_test.go]: Indeed, but I wanted to use a separate branch just for cleaning up tests (including using testify).

File: feed_flat_test.go (hunk 2 above)
Comment (reifcode) [inline feed_flat_test.go]: ok

File: activity_test.go (hunk 1 above)
Comment (JelteF) [inline activity_test.go]: Alright. I just meant that I added some testify tests already myself, so if you're writing new ones anyway it's nicer not to have to refactor them in the future already.

**Pull request merged**